src/test/java/edu/ucsb/cs156/happiercows/controllers/CommonsControllerTests.java

```diff
@@ -89,8 +89,50 @@ public void getCommonsTest() throws Exception {
     assertEquals(actualCommons, expectedCommons);
   }
 
+  //This common SHOULD be in the repository
   @WithMockUser(roles = { "USER" })
   @Test
+  public void getCommonsByIdTest_valid() throws Exception {
+    Commons Commons1 = Commons.builder()
+      .name("TestCommons2")
+      .id(18L)
+      .build();
+
+    when(commonsRepository.findById(eq(18L))).thenReturn(Optional.of(Commons1));
+    
+    MvcResult response = mockMvc.perform(get("/api/commons?id=18"))
+        .andExpect(status().isOk()).andReturn();
+
+    verify(commonsRepository, times(1)).findById(eq(18L));
+    String expectedJson = mapper.writeValueAsString(Commons1);
+    String responseString = response.getResponse().getContentAsString();
+    assertEquals(expectedJson, responseString);
+  }
+
+  //This common SHOULD NOT be in the repository
+  @WithMockUser(roles = { "USER" })
+  @Test
+  public void getCommonsByIdTest_invalid() throws Exception {
+
+    when(commonsRepository.findById(eq(18L))).thenReturn(Optional.empty());
+    
+    MvcResult response = mockMvc.perform(get("/api/commons?id=18"))
+        .andExpect(status().is(404)).andReturn();
+
+    verify(commonsRepository, times(1)).findById(eq(18L));
+    String responseString = response.getResponse().getContentAsString();
+
+    //Not sure how to expect this, seems like a mock json. Just parsed the string.
+    //For some reason they like to swap places randomly.
+    if(responseString.charAt(2) == 'm'){
+      assertEquals("{\"message\":\"Commons with id 18 not found\",\"type\":\"EntityNotFoundException\"}", responseString); 
+    }else{
+      assertEquals("{\"type\":\"EntityNotFoundException\",\"message\":\"Commons with id 18 not found\"}", responseString); 
+    }
```

Comment:
Can we use [`responseToJson`](https://github.com/ucsb-cs156-w22/team04-w22-6pm-HappyCows/blob/main/src/test/java/edu/ucsb/cs156/happiercows/ControllerTestCase.java#L34) here instead of testing different permutations of the JSON string?

Reply:
Apologies. If that's what that function made for, that's absolutely brilliant. Had no idea to look in that file.